index.js

```diff
@@ -560,10 +560,25 @@ function launchApp(argo, startExternalAdapterServer) {
 
         const startupAppOptions = convertOptions.getStartupAppOptions(configObject);
         const uuid = startupAppOptions && startupAppOptions.uuid;
+        const name = startupAppOptions && startupAppOptions.name;
         const ofApp = Application.wrap(uuid);
         const ofManifestUrl = ofApp && ofApp._configUrl;
         const isRunning = Application.isRunning(ofApp);
 
+        const { company, name: shortcutName } = shortcut;
+        let appUserModelId;
+        let namePart;
+
+        if (company) {
+            namePart = shortcutName ? `.${shortcutName}` : '';
+            appUserModelId = `${company}${namePart}`;
+        } else {
+            namePart = name ? `.${name}` : '';
+            appUserModelId = `${uuid}${namePart}`;
+        }
+
+        app.setAppUserModelId(appUserModelId);
```

Comment:
`BrowserWindow.setTaskbarGroup()` does some stuff with AppUserModelId. Can we make sure that these things play well together?

https://github.com/openfin/runtime/blob/develop/atom/browser/native_window_views_win.cc#L735

Reply:
To follow up to Sim's comment,  I did some quick tests and here are the findings:  Electron actually sets a default model Id with "electron.app.OpenFin", which explains why I saw notifications in a group called "electron.app.OpenFin" on Window server 2012.  So, we should set it to what we want.

app.setAppUserModelId does not seem to affect taskbar group code (link from Sim's comment)